backend/de.metas.business.rest-api-impl/src/test/java/de/metas/rest_api/v2/product/ProductRestControllerTest.java

```diff
@@ -0,0 +1,271 @@
+package de.metas.rest_api.v2.product;
+
+import ch.qos.logback.classic.Level;
+import de.metas.bpartner.BPartnerId;
+import de.metas.common.product.v2.response.JsonGetProductsResponse;
+import de.metas.common.product.v2.response.JsonProduct;
+import de.metas.common.product.v2.response.JsonProductBPartner;
+import de.metas.common.rest_api.JsonMetasfreshId;
+import de.metas.externalsystem.ExternalSystemConfigRepo;
+import de.metas.externalsystem.audit.ExternalSystemExportAuditRepo;
+import de.metas.logging.LogManager;
+import de.metas.product.ProductId;
+import de.metas.rest_api.externlasystem.dto.ExternalSystemService;
+import de.metas.uom.UomId;
+import de.metas.user.UserId;
+import de.metas.vertical.healthcare.alberta.dao.AlbertaProductDAO;
+import de.metas.vertical.healthcare.alberta.service.AlbertaProductService;
+import io.github.jsonSnapshot.SnapshotMatcher;
+import lombok.Builder;
+import lombok.NonNull;
+import org.adempiere.test.AdempiereTestHelper;
+import org.compiere.model.I_C_BPartner_Product;
+import org.compiere.model.I_C_UOM;
+import org.compiere.model.I_M_Product;
+import org.compiere.util.Env;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import static org.adempiere.model.InterfaceWrapperHelper.newInstance;
+import static org.adempiere.model.InterfaceWrapperHelper.saveRecord;
+import static org.assertj.core.api.Assertions.*;
+
+/*
+ * #%L
+ * de.metas.business.rest-api-impl
+ * %%
+ * Copyright (C) 2019 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+public class ProductRestControllerTest
```

Comment:
pls rename to ProductRest**s**ControllerTest..otherwise we might have problems finding it in future